research/H1/FPN.md

```diff
@@ -0,0 +1,106 @@
+# Label: H1, Preamble: "FPN"
+
+Flight Plan
+
+## Format
+
+The message begins with `FPN/` followed by a status. After that, it is a list of key/value pairs delimited by `:`. The final 4 characters are a checksum (algorithm unknown).
+
+### Known Statuses
+
+key | value
+---|-----
+RI | Route Inactive
+FNXXXX | Flight Number XXXX
+
+### Known Keys
+
+key | value
+---|-----
+A  | Arrival Procedure
+AA | Arrival Airport
+AP | Approach Procedure (?)
+CR | Current Route (?)
+D  | Departure Procedure
+DA | Departure Airport
+F  | First Waypoint
+R  | Departure Runway
+
+### First Waypoint
+There can be multiple First Waypoints in a message. I believe they are related to the prior keypair. see [Landing](#landing)
+
+### Waypoints format
+The use of the waypoints is designated by the prior key/value pair. 
+Waypoints can either be delimiated by `..`, `.` or `,`. The exact meaning is currently unclear. Current guess is as follows (with incorrect terminology)
+
+key | value
+---|-----
+.. | FAA Waypoint
+.  | Local
+,  | lat/lon in millidegrees
+
+## Examples
+
+### Landing
+https://app.airframes.io/messages/2159379444
+
+```
+FPN/RI:DA:KEWR:AA:KDFW:CR:EWRDFW01(17L)..SAAME.J6.HVQ.Q68.LITTR..MEEOW..FEWWW:A:SEEVR4.FEWWW:F:VECTOR..DISCO..RIVET:AP:ILS 17L.RIVET:F:TACKEC8B5
+```
+means the following:
+
+key | value
+---|-----
+Status | Route Inactive
+Departure Airport | KEWR
+Arrival Airport | KDWF
+Current Route | EWRDFW01(17L)..SAAME.J6.HVQ.Q68.LITTR..MEEOW..FEWWW
+Arrival Procedure | SEEVR4.FEWWW
```

Comment:
Do we want to further extract the arrival procedure + transition point?

Reply:
yes, I can extract it into `{procedureName},{transitionPoint}`

Reply:
I'm going to hold off on this, but update the docs, i think all procedures can be in the format `{name}.{waypoints}` where waypoints is the same format as routes